devfiles/che4z/devfile.yaml

```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: 1.0.0
+metadata:
+  generateName: che4z-
+components:
+  - 
+    type: chePlugin
+    id: broadcommfd/cobol-language-support/latest
+  - 
+    type: chePlugin
+    id: bitlang/cobol/latest
+  - 
+    type: chePlugin
+    id: broadcommfd/explorer-for-endevor/latest
+  - 
+    id: eclipse/che-machine-exec-plugin/7.3.0
+    type: chePlugin
+  - 
+    id: eclipse/che-theia/7.3.0
+    type: cheEditor
+  - 
+    alias: zowe-cli
+    type: kubernetes
+    mountSources: true
+    referenceContent: |
+         kind: List
```

Comment:
Is it possible to rewrite the devfile to follow the existing structure, or is the Kubernetes List required?

Reply:
Could you expand on what you refer to with 'existing structure'?  The Kubernetes List was required when compared to `dockerimage` type deployment, but I'm not sure if this is what you're referring to. We have persistence requirements for Zowe CLI profiles which required an additional PVC.

Reply:
In the past, using a Kubernetes List was necessary because the dockerimage format didn't support everything you might need, but now it's possible to specify volumes, environment vars, endpoints, etc. on a dockerimage component (e.g. for [java-spring](https://github.com/eclipse/che-devfile-registry/blob/master/devfiles/java-web-spring/devfile.yaml#L17-L35)).

I personally don't have a strong preference either way, but going this route would match the other devfiles.

Reply:
If there's no objection - I'd like to keep the devfile as kubernetes for this release and revisit the `dockerimage` format for a future revision. I think there's value in pursuing the `dockerimage` format in addition to alignment with the other devfiles. Which release of Che added the env, volumes, etc. support?

Reply:
That works for me, it's not like the referenceContent approach has any issues I can see -- it's mostly a consistency thing. I believe the additional fields (env, volumes, etc.) were in [the 7.0.0 release](https://github.com/eclipse/che/blob/7.0.0/wsmaster/che-core-api-workspace/src/main/resources/schema/devfile.json#L444) but our documentation somewhat lacking until relatively recently. 

For reference, the current documentation on devfiles is [here](https://www.eclipse.org/che/docs/che-7/making-a-workspace-portable-using-a-devfile/#component-type-dockerimage_making-a-workspace-portable-using-a-devfile). If you do decide to go down the path of using a `dockerimage` component, it'd be great to document any issues you encounter along the way.